Code/GraphMol/FilterCatalog/FilterMatchers.cpp

```diff
@@ -123,4 +123,29 @@ bool SmartsMatcher::hasMatch(const ROMol &mol) const {
             (d_max_count == UINT_MAX || count <= d_max_count));
   }
 }
+
+bool FilterHierarchyMatcher::getMatches(const ROMol &mol,
+                                        std::vector<FilterMatch> &m) const {
+  // a null matcher is root, just goes to the children
+
+  std::vector<FilterMatch> temp;
+  bool result = d_matcher->getMatches(mol, temp);
+
+  if (result) {
+    std::vector<FilterMatch> children;
+
+    for (size_t i = 0; i < d_children.size() && result; ++i) {
```

Comment:
Good catch, that was leftover logic from a previous incarnation.

----
Brian Kelley

> On Apr 25, 2016, at 11:01 PM, Greg Landrum <notifications@github.com> wrote:
> 
> In Code/GraphMol/FilterCatalog/FilterMatchers.cpp:
> 
> > @@ -123,4 +123,29 @@ bool SmartsMatcher::hasMatch(const ROMol &mol) const {
> >              (d_max_count == UINT_MAX || count <= d_max_count));
> >    }
> >  }
> > +
> > +bool FilterHierarchyMatcher::getMatches(const ROMol &mol,
> > +                                        std::vector<FilterMatch> &m) const {
> > +  // a null matcher is root, just goes to the children
> > +
> > +  std::vector<FilterMatch> temp;
> > +  bool result = d_matcher->getMatches(mol, temp);
> > +
> > +  if (result) {
> > +    std::vector<FilterMatch> children;
> > +
> > +    for (size_t i = 0; i < d_children.size() && result; ++i) {
> better to use BOOST_FOREACH here.
> The && result isn't doing anything
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>